tests/components/ecovacs/test_sensor.py

```diff
@@ -152,3 +152,31 @@ async def test_disabled_by_default_sensors(
         ), f"Entity registry entry for {entity_id} is missing"
         assert entry.disabled
         assert entry.disabled_by is er.RegistryEntryDisabler.INTEGRATION
+
+
+@pytest.mark.usefixtures(
+    "entity_registry_enabled_by_default", "mock_vacbot", "init_integration"
+)
+@pytest.mark.parametrize(("device_fixture"), ["123"])
+async def test_legacy_sensors(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    mock_vacbot,
+) -> None:
+    """Test that sensor entity snapshots match."""
+    mock_vacbot.components = {"main_brush": 0.8, "side_brush": 0.6, "filter": 0.4}
+    hass.bus.async_fire("ecovacs_legacy_lifespan_event")
```

Comment:
Can we test this without using an event to fire? It feels like were indirectly touching internals and that feels strange

Reply:
tbh, this was the easiest way :see_no_evil:
it might be possible to do without firing an event, but this would need a (_maybe complete_) mock of the `EcoVacsXMPP` from the `sucks` lib :thinking:

Reply:
You can use AsyncMocks right?

Reply:
unfortunately, the sucks lib is not async, but regular Mock could be used (_and it is already used for the `mock_vacbot`_), but at the moment it does not mock the event system of the `VacBot` at all. So to avoid the firing of events in HA, we need first to complete mock the event system of the `VacBot` class

Reply:
5048864